src/libraries/Common/src/System/Security/Cryptography/RSAOpenSsl.cs

```diff
@@ -555,6 +513,30 @@ private void ThrowIfDisposed()
             }
         }
 
+        private SafeEvpPKeyHandle GetPKey()
+        {
+            SafeRsaHandle currentKey = GetKey();
+            SafeEvpPKeyHandle pkeyHandle = Interop.Crypto.EvpPkeyCreate();
+
+            try
+            {
+                // Wrapping our key in an EVP_PKEY will up_ref our key.
+                // When the EVP_PKEY is Disposed it will down_ref the key.
+                // So everything should be copacetic.
+                if (!Interop.Crypto.EvpPkeySetRsa(pkeyHandle, currentKey))
+                {
+                    throw Interop.Crypto.CreateOpenSslCryptographicException();
+                }
+
+                return pkeyHandle;
```

Comment:
Nit: consider moving this to after the catch block.  The code is fine as-is, but my eyes were initially confused by the dispose below, until I realized it was a catch block rather than finally.  Not a big deal.